server/backend/api/app/routers/router_guards.py

```diff
@@ -30,25 +40,68 @@
 ERROR_EXTERNAL_USER_ACTION_PROHIBITED = "external_user_action_prohibited"
 
 no_requester_exception = HTTPException(
-    status_code=HTTPStatus.FORBIDDEN, # 403
+    status_code=HTTPStatus.FORBIDDEN,  # 403
     detail={
         "code": ERROR_REQUESTER_NOT_EXISTS,
         "description": "Requester does not exist, action is not allowed",
-    }
+    },
 )
 
 external_user_prohibited_exception = HTTPException(
     status_code=HTTPStatus.FORBIDDEN,
     detail={
         "code": ERROR_EXTERNAL_USER_ACTION_PROHIBITED,
         "description": "Action is not allowed for external user.",
-    }
+    },
 )
 
+
+async def get_current_requester(
+    request_cognito_user_id: str = Depends(get_request_cognito_user_id),
+    db: Session = Depends(database.get_db),
+):
+    fam_user: FamUser = crud_user.get_user_by_cognito_user_id(
+        db, request_cognito_user_id
+    )
+    if fam_user is None:
+        raise no_requester_exception
+
+    requester = Requester.model_validate(fam_user)
+    LOGGER.debug(f"Current request user (requester): {requester}")
+    return requester
+
+
+def authorize(
+    claims: dict = Depends(validate_token),
+    db: Session = Depends(database.get_db),
+    requester: Requester = Depends(get_current_requester),
+):
```

Comment:
the new authorize method is used for the general authorization check, used by Forest Client API integration and IDIM proxy integration, so user needs to have at least one admin or delegated admin role in order to call these integration apis

(fyi, this is the first level of security check, we'll add organization check later)

Reply:
I feel like some of these comments of yours should be put as comments on the code.

Reply:
sure, I'll add some comment in the code!